platform/ios/src/MGLMapView.mm

```diff
@@ -4577,12 +4577,17 @@ - (void)setSelectedAnnotations:(NSArray<id <MGLAnnotation>> *)selectedAnnotation
 
 - (void)selectAnnotation:(id <MGLAnnotation>)annotation animated:(BOOL)animated
 {
-    MGLLogDebug(@"Selecting annotation: %@ animated: %@", annotation, MGLStringFromBOOL(animated));
+    [self selectAnnotation:annotation moveIntoView:animated animateSelection:animated];
+}
+
+- (void)selectAnnotation:(id <MGLAnnotation>)annotation moveIntoView:(BOOL)moveIntoView animateSelection:(BOOL)animateSelection
+{
+    MGLLogDebug(@"Selecting annotation: %@ moveIntoView: %@ animateSelection: %@", annotation, MGLStringFromBOOL(moveIntoView), MGLStringFromBOOL(animateSelection));
     CGRect positioningRect = [self positioningRectForAnnotation:annotation defaultCalloutPoint:CGPointZero];
-    [self selectAnnotation:annotation moveOnscreen:animated animateSelection:YES calloutPositioningRect:positioningRect];
+    [self selectAnnotation:annotation moveIntoView:moveIntoView animateSelection:animateSelection calloutPositioningRect:positioningRect];
```

Comment:
Should deselection in response to another annotation being selected also honor the `animated` flag, for consistency?

https://github.com/mapbox/mapbox-gl-native/blob/e2e38dd7af4a21bf46039bf5434444ac6efbbdcd/platform/ios/src/MGLMapView.mm#L4596
https://github.com/mapbox/mapbox-gl-native/blob/e2e38dd7af4a21bf46039bf5434444ac6efbbdcd/platform/ios/src/MGLMapView.mm#L4882

Reply:
Thanks for the quick review!

I had the very same question 🙂 I can imagine situations where different applications would want different results. I think leaving as is makes the most sense at present, and we can tackle in a subsequent PR if necessary.